### PR TITLE
MM-32525 added unread_mentions_per_channel

### PR DESCRIPTION
**Pull request by reflog**
#### Ticket Link

https://mattermost.atlassian.net/browse/MM-32525

### CODE REVIEW

File: src/reducers/entities/threads.ts [hunk 1]
```diff
@@ -87,10 +87,20 @@ export const threadsInTeamReducer = (state: ThreadsState['threadsInTeam'] = {},
 };
 export const countsReducer = (state: ThreadsState['counts'] = {}, action: GenericAction) => {
     switch (action.type) {
+    case ThreadTypes.RECEIVED_PER_CHANNEL_MENTION_COUNTS: {
+        return {
+            ...state,
+            [action.data.team_id]: {
+                ...state[action.data.team_id]?.unread_mentions_per_channel ?? {},
```
Comment (calebroseland) [inline src/reducers/entities/threads.ts:94]:
```suggestion
                ...state[action.data.team_id] ?? {},
```

File: src/types/threads.ts [hunk 2]
```diff
@@ -31,5 +31,6 @@ export type ThreadsState = {
         total: number;
         total_unread_threads: number;
         total_unread_mentions: number;
+        unread_mentions_per_channel: Record<string, number>;
```
Comment (calebroseland) [inline src/types/threads.ts:34]: `import type {Channel} from './channels';`
```suggestion
        unread_mentions_per_channel: Record<$ID<Channel>, number>;
```

File: src/reducers/entities/threads.test.js [hunk 3]
```diff
@@ -6,73 +6,92 @@ import deepFreeze from 'utils/deep_freeze';
...
-                    team_id: 'a',
-                    threads: [
-                        {id: 't1'},
-                    ],
-                    total: 3,
-                    total_unread_threads: 0,
-                    total_unread_mentions: 1,
-                },
-            });
+    test('RECEIVED_THREADSshould  update the state', () => {
```
Comment (calebroseland) [inline src/reducers/entities/threads.test.js:9]:
```suggestion
    test('RECEIVED_THREADS should update the state', () => {
```

File: src/client/client4.ts [hunk 4]
```diff
@@ -1941,6 +1941,14 @@ export default class Client4 {
         );
     };
 
+    getThreadMentionCountsByChannel = (userId: string, teamId: string) => {
+        const url = `${this.getUserThreadsRoute(userId, teamId)}`;
```
Comment (calebroseland) [inline src/client/client4.ts:1945]:
```suggestion
        const url = `${this.getUserThreadsRoute(userId, teamId)}/mention_counts`;
```

**Review by calebroseland — CHANGES_REQUESTED** [review]

**Review by jespino — APPROVED** [review]
LGTM, except for the @calebroseland comments.

**Review by calebroseland — APPROVED** [review]

**Pull request merged**